presto-spi/src/main/java/io/prestosql/spi/connector/ConnectorSplitManager.java

```diff
@@ -34,6 +38,16 @@ default ConnectorSplitSource getSplits(
         throw new UnsupportedOperationException();
     }
 
+    default ConnectorSplitSource getSplits(
+            ConnectorTransactionHandle transaction,
+            ConnectorSession session,
+            ConnectorTableHandle table,
+            SplitSchedulingStrategy splitSchedulingStrategy,
+            Supplier<TupleDomain<ColumnHandle>> dynamicFilter)
```

Comment:
I think SPI will change eventually to support lazy dynamic filters. (FYI: @martint and @rzeyde-varada )